app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Abilities for all users
+    can :read, :all
+
+    # Abilities for authenticated users
+    return unless user.present?
+
+    can :manage, Post, author_id: user.id
+    can :create, Comment
+    can :manage, Comment, user_id: user.id
+    can :destroy, Like, user_id: user.id
+
+    # Abilities for admin users
+    can :manage, :all if user.admin?
+  end
+end
```

Comment:
- [ ] @Team Great work adding the Ability class to add authorization, however, I could not find the CanCan gem in your gemfile which I suspect is leading to the error below, CanCan gem is part of the Ability class and is used for authorization in Ruby on Rails applications. Kindly consider adding the Gem.
![image](https://github.com/the-faizmohammad/BlogApp/assets/61408860/8783d9bd-51ae-4132-8b71-2436b7bbefe9)